anilistUpdater/anilistUpdater.py

```diff
@@ -295,72 +408,22 @@ def handle_filename(self, filename):
-                    self.cache_to_file(filename, file_info.get('name'), result)
+        cache_entry = self.check_and_clean_cache(filename, file_info.get('name'))
 
-        # True for opening AniList and updating next episode.
-        else:
-            print(f'Found in cache! {cached_result}')
-            # Only proceed if cached_result is a tuple/list and has enough elements
-            if isinstance(cached_result, (tuple, list)) and len(cached_result) >= 4:
-                # Change to the episode that needs to be updated
-                if len(cached_result) > 5:
-                    cached_result = tuple(cached_result[:4]) + (file_info.get('episode'),) + tuple(cached_result[5:])
-                else:
-                    cached_result = tuple(cached_result[:4]) + (file_info.get('episode'),)
-                # Ensure tuple is 6 elements (pad with "CURRENT" if needed)
-                if len(cached_result) == 5:
-                    cached_result = cached_result + ("CURRENT",)
-                result = self.update_episode_count(cached_result)
-
-                # If it's different, update in cache as well.
-                if cached_result != result and result:
-                    print(f'Updating cache to: {result}')
-                    self.update_cache(filename, file_info.get('name'), result, line_index)
-
-                # If it either errored or couldn't update, retry without cache.
-                if not result:
-                    print('Failed to update through cache, retrying without.')
-                    # Deleting from the cache
-                    self.update_cache(filename, file_info.get('name'), None, line_index)
-                    # Retrying
-                    self.handle_filename(filename)
-            else:
-                print('Cached result is invalid, ignoring cache.')
-                # Remove invalid cache entry
-                if line_index is not None:
-                    self.update_cache(filename, file_info.get('name'), None, line_index)
-                # Retry without cache
-                self.handle_filename(filename)
+        # If launching and cache has anime_id, we can skip search and open directly.
+        if self.ACTION == 'launch' and cache_entry and cache_entry.get('anime_id'):
+            anime_id = cache_entry['anime_id']
+            print(f'Opening AniList (cached) for guessed "{file_info.get("name")}": https://anilist.co/anime/{anime_id}')
+            webbrowser.open_new_tab(f'https://anilist.co/anime/{anime_id}')
+            return
+
+        # Always fetch fresh info (simplified due to reduced cache fields)
+        result = self.get_anime_info_and_progress(file_info.get('name'), file_info.get('episode'), file_info.get('year'))
+        result = self.update_episode_count(result)
 
+        if result and result[2] is not None:
+            # Update structured cache
+            self.cache_to_file(filename, file_info.get('name'), result)
```

Comment:
The key 'last_progress' is also unused due to the reduced cache fields. This leaves us with a far more simplified but costs us double the API calls

You can probably find a way to use the same approach as before (current_progress, total_episodes, file_progress, current_status) or a better one